content/en/tracing/trace_collection/automatic_instrumentation/dd_libraries/nodejs.md

```diff
@@ -136,13 +136,50 @@ esbuild.build({
   outfile: 'out.js',
   plugins: [ddPlugin],
   platform: 'node', // allows built-in modules to be required
-  target: ['node16']
+  target: ['node16'],
+  external: [
+    // esbuild cannot bundle native modules
+    '@datadog/native-metrics',
+
+    // required if you use profiling
+    '@datadog/pprof'
+
+    // required if you use Datadog security features
+    '@datadog/native-appsec',
+    '@datadog/native-iast-taint-tracking',
+    '@datadog/native-iast-rewriter'
+
+    // required if you encounter graphql errors during the build step
+	'graphql/language/visitor',
+    'graphql/language/printer',
+    'graphql/utilities',
+  ]
 }).catch((err) => {
   console.error(err)
   process.exit(1)
 })
 ```
 
+Note that due to the usage of native modules in the tracer, which are compiled C++ code in a usually ending with a `.node` file extension, you'll need to add entries to your `external` list. Currently native modules used in the Node.js tracer live inside of `@datadog` prefixed packages. This will also require that you ship a `node_modules/` directory alongside your bundled application. You probably don't want to ship your entire `node_modules/` directory as it would contain many superfluous packages that should be contained in your bundle.
```

Comment:
```suggestion
**Note**: Due to the usage of native modules in the tracer, which are compiled C++ code, (usually ending with a `.node` file extension), you need to add entries to your `external` list. Currently native modules used in the node.js tracer live inside of `@datadog` prefixed packages. This will also require that you ship a `node_modules/` directory alongside your bundled application. You don't need to ship your entire `node_modules/` directory as it would contain many superfluous packages that should be contained in your bundle.
```